src/modules/service-provider/domain/service-provider.service.ts

```diff
@@ -81,6 +104,58 @@ export class ServiceProviderService {
         return Array.from(serviceProviders.values());
     }
 
+    public async getOrganisationRollenAndRollenerweiterungenForServiceProviders(
+        serviceProviders: ServiceProvider<true>[],
+    ): Promise<ManageableServiceProviderWithLinkedObjects[]> {
+        const serviceProvidersIds: ServiceProviderID[] = serviceProviders.map((sp: ServiceProvider<true>) => sp.id);
+        const rollen: Map<ServiceProviderID, Rolle<true>[]> =
+            await this.rolleRepo.findByServiceProviderIds(serviceProvidersIds);
+        const rollenerweiterungen: Map<ServiceProviderID, Rollenerweiterung<true>[]> =
+            await this.rollenerweiterungRepo.findByServiceProviderIds(serviceProvidersIds);
+        const organisationen: Map<ServiceProviderID, Organisation<true>> = await this.organisationRepo.findByIds(
+            serviceProviders.map((sp: ServiceProvider<true>) => sp.providedOnSchulstrukturknoten),
+        );
+
+        return serviceProviders.map((serviceProvider: ServiceProvider<true>) => ({
+            serviceProvider,
+            organisation: organisationen.get(serviceProvider.providedOnSchulstrukturknoten)!,
+            rollen: rollen.get(serviceProvider.id) ?? [],
+            rollenerweiterungen: rollenerweiterungen.get(serviceProvider.id) ?? [],
+        }));
+    }
+
+    public async getRollenerweiterungenForDisplay(
+        rollenerweiterungen: Rollenerweiterung<true>[],
+    ): Promise<RollenerweiterungForManageableServiceProvider[]> {
+        const organisationen: Map<OrganisationID, Organisation<true>> = await this.organisationRepo.findByIds(
+            rollenerweiterungen.map((rollenerweiterung: Rollenerweiterung<true>) => rollenerweiterung.organisationId),
+        );
+        const rollen: Map<RolleID, Rolle<true>> = await this.rolleRepo.findByIds(
+            rollenerweiterungen.map((rollenerweiterung: Rollenerweiterung<true>) => rollenerweiterung.rolleId),
+        );
+        const serviceProvider: Map<ServiceProviderID, ServiceProvider<true>> = await this.serviceProviderRepo.findByIds(
+            rollenerweiterungen.map(
+                (rollenerweiterung: Rollenerweiterung<true>) => rollenerweiterung.serviceProviderId,
+            ),
+        );
+
+        return rollenerweiterungen.map((rollenerweiterung: Rollenerweiterung<true>) => ({
+            organisation: {
+                id: rollenerweiterung.organisationId,
+                name: organisationen.get(rollenerweiterung.organisationId)?.name ?? '',
+                kennung: organisationen.get(rollenerweiterung.organisationId)?.kennung,
+            },
+            rolle: {
+                id: rollenerweiterung.rolleId,
+                name: rollen.get(rollenerweiterung.rolleId)?.name ?? '',
+            },
+            serviceProvider: {
+                id: rollenerweiterung.serviceProviderId,
+                name: serviceProvider.get(rollenerweiterung.serviceProviderId)?.name ?? '',
+            },
+        }));
+    }
```

Comment:
Reduces redundant lookups:

```suggestion
return rollenerweiterungen.map(
  (rollenerweiterung): RollenerweiterungForManageableServiceProvider => {
    const org: Organisation<true> = organisationen.get(rollenerweiterung.organisationId);
    const rolle: Rolle<true>= rollen.get(rollenerweiterung.rolleId);
    const sp: Serviceprovider<true> = serviceProvider.get(rollenerweiterung.serviceProviderId);

    return {
      organisation: {
        id: rollenerweiterung.organisationId,
        name: org?.name ?? '',
        kennung: org?.kennung,
      },
      rolle: {
        id: rollenerweiterung.rolleId,
        name: role?.name ?? '',
      },
      serviceProvider: {
        id: rollenerweiterung.serviceProviderId,
        name: sp?.name ?? '',
      },
    };
  },
);

```